src/tests/radmin/all.mk

```diff
@@ -0,0 +1,148 @@
+#
+#	Unit tests for radmin
+#
+
+#
+#	Test name
+#
+TEST := test.radmin
+
+#
+#	Config settings
+#
+PORT               := 12341
+BIN_PATH           := $(BUILD_DIR)/bin/local
+
+RADMIN_BIN         := $(BIN_PATH)/radmin
+RADMIN_OUTPUT_DIR  := $(BUILD_DIR)/tests/radmin
+RADMIN_RADIUS_LOG  := $(RADMIN_OUTPUT_DIR)/radius.log
+RADMIN_GDB_LOG     := $(RADMIN_OUTPUT_DIR)/gdb.log
+RADMIN_SOCKET_FILE := $(RADMIN_OUTPUT_DIR)/control-socket.sock
+RADMIN_TEST_PATH   := src/tests/radmin
+RADMIN_CMDS_PATH   := $(RADMIN_TEST_PATH)/cmds
+RADMIN_CONFIG_PATH := $(RADMIN_TEST_PATH)/config
+
+#
+#	The test files are files without extensions.
+#
+RADMIN_TEST_FILES := $(patsubst $(RADMIN_CMDS_PATH)/%.in,$(RADMIN_CMDS_PATH)/%,$(wildcard $(RADMIN_CMDS_PATH)/*.in))
+RADMIN_OK_FILES   := $(patsubst $(RADMIN_CMDS_PATH)/%,$(RADMIN_OUTPUT_DIR)/%.ok,$(RADMIN_TEST_FILES))
+
+#
+#	Create the output directory
+#
+.PHONY: $(RADMIN_OUTPUT_DIR)
+$(RADMIN_OUTPUT_DIR):
+	${Q}mkdir -p $@
+
+#
+#	Main
+#
+
+# This gets called recursively, so has to be outside of the condition below
+# We can't make this depend on radiusd.pid, because then make will create
+# radiusd.pid when we make radiusd.kill, which we don't want.
+.PHONY: test.radmin.kill
+test.radmin.kill: | $(RADMIN_OUTPUT_DIR)
+	${Q}if [ -f $(RADMIN_CONFIG_PATH)/radiusd.pid ]; then \
+		ret=0; \
+		if ! ps `cat $(RADMIN_CONFIG_PATH)/radiusd.pid` >/dev/null 2>&1; then \
+		    rm -f $(RADMIN_CONFIG_PATH)/radiusd.pid; \
+		    echo "FreeRADIUS terminated during test"; \
+		    echo "GDB output was:"; \
+		    cat "$(RADMIN_GDB_LOG)"; \
+		    echo "--------------------------------------------------"; \
+		    tail -n 40 "$(RADMIN_RADIUS_LOG)"; \
+		    echo "Last entries in server log ($(RADMIN_RADIUS_LOG)):"; \
+		    ret=1; \
+		fi; \
+		if ! kill -TERM `cat $(RADMIN_CONFIG_PATH)/radiusd.pid` >/dev/null 2>&1; then \
+			ret=1; \
+		fi; \
+		exit $$ret; \
+	fi
+
+.PHONY: clean.test.radmin
+clean.test.radmin:
+	${Q}rm -f $(RADMIN_OUTPUT_DIR)/*.ok \
+		  $(RADMIN_OUTPUT_DIR)/*.log \
+		  $(RADMIN_OUTPUT_DIR)/*.skip \
+		  $(RADMIN_CONFIG_PATH)/test.conf \
+		  $(RADMIN_SOCKET_FILE)
+
+clean.test: clean.test.radmin
+
+#
+#	Build the configuration files.
+#
+$(RADMIN_CONFIG_PATH)/dictionary:
+	${Q}echo "# test dictionary not install.  Delete at any time." > $@
+
+$(RADMIN_CONFIG_PATH)/test.conf: $(RADMIN_CONFIG_PATH)/dictionary
```

Comment:
This should just be a static configuration file in src/tests/radmin

The main server binary takes the directory from `-d whatever`, and creates a configuration item of `raddbdir = whatever`.  The test of the configuration items can be taken from that:

```
testdir = ${raddbdir}/config
logdir = ${raddbdir}/../../build/src/tests/radmin
```
etc.  That simplifies things a fair bit.